### ISSUE TITLE
BLMASK  filtering

### ISSUE THREAD

**Issue by kaciker** [dragino/TrackerD#28]
Hello,

I'm having problems using the BLMASK filtering.

I need to filter the UUIDs of the BCN05 beacons that we installed so that we do not receive UUIDs from other  (BLE) devices in the area from other projects.

If I deactivate the BLMASK by setting it to 0, the tracker gives me the strongest UUID, but sometimes it provides UUIDs from other devices that are not BCN05, which disrupts the project.

To address this, I tried to set the BLMASK. According to the Dragino documentation, the BLMASK needs to be between 6 and 10 bytes. The standard UUID for the Dragino BCN05 is 0112233445566778899aabbccddeeff0, which is 16 bytes. However, the sensor does not allow me to set BLMASK to 0112233445566778899aabbccddeeff0 and only allows me to set 6 bytes. With this 6-byte setting,AT+BLEMASK=0112233445 filtering does not work, and as a result, the GPS is activated all the time, and the BLE iBeacon functionality ceases to work.

What am I doing wrong?

**Comment by namejinnameyuan** [comment]
The BLMASK setting filters out the beacons you don’t need, such as UUID: 0112233445566778899aabbccddeeff0. If you don’t need this type of beacon, then set BLMASK to 011225, and the beacon will not be scanned. If the RSSI of the beacon exceeds -90, it will also be filtered out.
sc_count = 0 This means there is no useful beacon scanned

**Comment by kaciker** [comment]
> The BLMASK setting filters out the beacons you don’t need, such as UUID: 0112233445566778899aabbccddeeff0. If you don’t need this type of beacon, then set BLMASK to 011225, and the beacon will not be scanned. If the RSSI of the beacon exceeds -90, it will also be filtered out. sc_count = 0 This means there is no useful beacon scanned

Hello
 I don't know where is the mistake, in your answer or in the dragino doc.

"...3.2.12  Set BLEMASK to filter BLE iBeacon
BLEMASK is to filter the unwanted BLE iBeacons during scan. **For example, if BLEMASK is 123456. TrackerD will only uplink UUID info which includes 123456**. **It will ignore all other iBeacons** which doesn’t contact 123456 in the UUID.

Note: BLEMASK range is 6 ~ 10 bytes. If AT+BLEMASK < 6 bytes, BLEMASK will be disabled.

**Comment by namejinnameyuan** [comment]
Yes, filter out beacons other than BLEMASK, and set the length to 6~10.

**Comment by namejinnameyuan** [comment]
You can use AT+SHOWID=1 to see the scanning information. If the signal strength of the beacon is lower than -90, it will be abandoned.
